### DISCUSSION TITLE
People Search - Search Scope (Source)

### DISCUSSION THREAD

**Discussion by rehkech** [microsoft-search/pnp-modern-search#3648] [Q&A]
* This is not a feature request but the page at https://microsoft-search.github.io/pnp-modern-search/#issues-questions-feedback says that questions can be posted via the Git Hub route. My apologies if this is in the wrong place. Please can this be moved if needed *

Hi everyone,

I'm looking for some advice from anyone that has experience of the SharePoint '[pnp Modern Search' (v4) tool](https://microsoft-search.github.io/pnp-modern-search/).

We have a requirement to build a people search which does not search across the tenant 'GAL', but instead would search to a more localised scope, for example a SharePoint list within the scope of the site hosting the pnp Modern Search page.

I had a look at the documentation https://microsoft-search.github.io/pnp-modern-search/scenarios/Create-a-useful-People-Search/ and in the section 'Configure Search Results' there is mention of a 'Result Source ID' (https://microsoft-search.github.io/pnp-modern-search/scenarios/Create-a-useful-People-Search/#configure-search-results) which appears to be where the scope is defined. 

Unfortunately I can't find more detail than this in the documentation, IE to specify what other sources would be acceptable and if these include a local SharePoint list.

I would like to try and determine the answer for this before configuring and deploying only to find the required functionality is not available.

Thanks very much for your help!

**Comment by kasperbolarsen** [comment]
In my opinion you are moving away from people search once your data source no longer is localPeopleResults or the equivalent in Graph.
It is possible to set up a SharePoint list to contain fields like name, email, department and so on, and display these data in a similar way as people search, but the data will be in localSharePointResults as the source is a list and not the User Profile Application.

Could you share the purpose of this search, is it another version of the People web part or are you intension to show everybody with access to the site ?

**Reply by rehkech** [reply]
Hi @kasperbolarsen , many thanks for your response and insight.

I'm yet to meet the users for detailed requirements gathering, but my understanding at a high level is that the users currently have two personnel lists; one of these is outside the tenant and one is a subset of information within the tenant. I am unsure if the two resources sync with any 'centralised' resource(s) at the moment and if so, what access to this would entail. 

I believe the user has determined there is insufficient criteria within the person 'object' for the list within the host tenant to 'filter' by only those people on the tenant list of identities who should fit on to their desired list. For example, it is not as simple as everyone in 'Department A' should be on their list and no one else should appear. In reality there are people from Departments A, B and C, but only some people in Department C should be on the personnel list but there are insufficient criteria to define this 'programmatically'.

The user wants to 'amalgamate' the two lists (outside tenant and selected people from within the tenant). The list should then be made available to all users enrolled in the SharePoint site (where presumably the list will be hosted).

One final note - it is extremely unlikely I would be able to affect any aspect of identity within the tenant. For example, say there was a custom field on the person 'object' that could be added or edited to those people in scope of the list, I doubt I would be able to implement this.

As I said this is a very early, high level review, which I am sure will be refined once requirements gathering gets underway - but I'm absolutely open to any feedback or steers that would be suggested. It's certainly a concern that the duplication of data (within the tenant, at least) to refine it down to a "localised" list would be the only way to achieve this.

Thanks very much again!
